server/src/main/java/io/spine/server/entity/storage/Introspector.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.entity.storage;
+
+import com.google.common.collect.ImmutableMap;
+import io.spine.base.EntityWithColumns;
+import io.spine.code.proto.FieldDeclaration;
+import io.spine.server.entity.model.EntityClass;
+import io.spine.server.entity.storage.InterfaceBasedColumn.GetterFromEntity;
+import io.spine.server.entity.storage.InterfaceBasedColumn.GetterFromState;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.spine.code.proto.ColumnOption.columnsOf;
+import static io.spine.reflect.Methods.setAccessibleAndInvoke;
+import static io.spine.util.Exceptions.newIllegalStateException;
+
+/**
+ * An extractor of entity columns.
+ */
+final class Introspector {
```

Comment:
An **intro**spector examines its own properties. This is an `Inspector`, or maybe a `Scanner`.